lib/DxilValidation/DxilValidation.cpp

```diff
@@ -970,6 +970,175 @@ static void ValidateImmOperandForMathDxilOp(CallInst *CI, DXIL::OpCode opcode,
   }
 }
 
+static bool CheckFromRegisterInterpretations(uint32_t Ri) {
+  std::set<DXIL::ComponentType> ValidSet = {
```

Comment:
It would almost certainly be more efficient to implement this by having a static array of component types and then doing a linear search through it to see if Ri is in that array.

As it is written now (and for all of the `std::set` based ones, this is building up dynamically allocated complex data structures and throwing them away for each call.

Reply:
yeah! I hadn't considered that even for static initialization the memory is dynamically allocated.

Reply:
Catching up. This has been resolved.